### PR TITLE
Scrape

### LINKED ISSUE #1: Discussion: JSON data
It would be nice if the JSON includes nepali date(BS), english date (AD), tithi, etc.

### PR DESCRIPTION
**Pull request by paudsu01**
Scrape hamropatro's website and store the provided year's calendar data in JSON files

### CODE REVIEW

File: .gitignore
Comment (sumanchapai) [inline .gitignore:1]: Add `.DS_Store`

File: scrape/scrape.py
Comment (sumanchapai) [inline scrape/scrape.py:1]: Use `isort` to sort imports

File: scrape/scrape.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    max_month : int = 12
+
+    all_months_data : Dict[str, Dict] = {}
+
+    while current_month <= max_month:
+
+        current_month_dict : Dict[str, Union[str,Dict]] = {}
+        all_months_data[INT_TO_MONTH[current_month]] = current_month_dict
+
+        r : requests.models.Response = requests.get(url=HAMRO_PATRO_URL+str(current_month))
```
Comment (sumanchapai) [inline scrape/scrape.py]: Handle exception properly when the year is invalid like negative number or `1000` or something like that… In general, you should check if the response status code is not 200.

File: scrape/scrape.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import sys
+import os
+import datetime
+import time
+import bs4
+
+# Example: python scrape.py 2080 fetches data for 2080B.S and saves them in database/2080.json
+
+def fetch_and_parse(year : int) -> Dict[str, Dict]:
+
```
Comment (sumanchapai) [inline scrape/scrape.py:33]: Use a formatter like `yapf` to enforce consistent formatting. Also add `yapf` configuration file in the root.

File: scrape/scrape.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    year : str = sys.argv[1]
+    if not year.isnumeric():
+        raise NonNumericYearArgumentException("Year argument needs to be numeric")
+    
+    response_dict : Dict[ str, Union[Dict, int]] = {}
+    response_dict["months"] : Dict[str, Dict] = fetch_and_parse(int(year))
+    response_dict["year"] : int = int(year)
+    response_dict["meta"] : Dict[str, str] = {"year_format":"YYYY-MM-DD"}
+    save_as_json_file(response_dict, year)
```
Comment (sumanchapai) [inline scrape/scrape.py:184]: Also, please save the JSON in a directory called `data` in the project root.

File: scrape/scrape.py (hunk 1 above)
Comment (paudsu01) [reply scrape/scrape.py]: The negative year values are handled by the `isnumeric` string method. https://github.com/sumanchapai/patro/blob/72bc85b4032c835f26e2d5d48b277987437f8157/scrape/scrape.py#L111-L112
However, I will modify to check for other invalid year values with response status code which I agree is something that needs to be handled.

**Review by sumanchapai — APPROVED** [review]

**Comment by paudsu01** [comment]
Resolved all changes here + the ones mentioned in JSON data issue [here](https://github.com/sumanchapai/patro/issues/1). Please review the changes @sumanchapai

**Pull request merged**